src/lib/data/index.ts

```diff
@@ -268,3 +270,189 @@ export function useSettings(): {
 
     return { settings, updateSettings, loadSettings, handlers };
 }
+
+export interface AlertList {
+    [date: string]: string[];
+}
+
+export const defaultAlertList = {};
+
+export function useAlertList(): {
+    alertList: AlertList;
+    addAlert: (date: string, id: string) => void;
+    removeAlert: (date: string, id: string) => void;
+    removeDateKey: (date: string) => void;
+    loadAlertList: () => Promise<void>;
+    pollAlertList: (eventsQuery: QueryResult<EventListQuery, EventListQueryVariables>, interval?: number) => void;
+} {
+    const storage = useStorage();
+    const [alertList, setAlertList] = useState<AlertList>(defaultAlertList);
+
+    // The time of the event(s) has come (most likely),
+    // so we are being told to remove the dynamic date (as an obj property), as well as from the dateKeys array
+    const removeDateKey = useCallback(
+        (date: string) => {
+            setAlertList((state) => {
+                const cleanState: AlertList = { ...state };
+                if (cleanState[date]) {
+                    delete cleanState[date];
+                }
+                try {
+                    void storage.put('alertList', JSON.stringify(cleanState));
+                } catch (_error) {
+                    // Nothing to do here, it just wont save
+                }
+                return cleanState;
+            });
+        },
+        [alertList, storage, setAlertList]
```

Comment:
Does `storage` need to be a dependency here since you're not reading anything from it?

Reply:
I believe so because I dont want to be using a stale storage obj

Reply:
Yeah but does it matter if you're just updating the `alertList` prop and not reading anything from it?

Reply:
I'm updating the storage in the trycatch

Reply:
Right but you're not appending to the list or anything, just overwriting the `alertList` prop in storage. So if `storage` changes it doesn't matter, cause you're not reading anything from it, just writing. From my understanding of the `useCallback` hook, the memoized function would change if any of the deps change, but in this case `storage` doesn't seem to matter in how you're using it.

Reply:
👍